mmyolo/models/task_modules/assigners/batch_task_aligned_assigner.py

```diff
@@ -43,14 +46,17 @@ def __init__(self,
                  alpha: float = 1.0,
                  beta: float = 6.0,
                  eps: float = 1e-7,
-                 use_ciou: bool = False):
+                 use_ciou: bool = False,
+                 gpu_assign_thr: float = -1):
```

Comment:
Thank you very much for your contribution. May I ask how this parameter is set more reasonable by the user? Or is there a rule of thumb?

Reply:
Can we find a way to make it more user-friendly and automated?

Reply:
> Thank you very much for your contribution. May I ask how this parameter is set more reasonable by the user? Or is there a rule of thumb?

In general, users are unlikely to need to adjust this parameter. However, in situations like the Widerface dataset where an image contains a large number of bounding boxes, this parameter becomes important. The value is chosen with the goal of utilizing GPU processing as much as possible, and only switching to CPU when out-of-memory errors occur. In my experience with a setup of 6x 1080ti GPUs, setting gpu_assign_thr=300 approaches the upper limit of GPU usage.

batch_size = 16

backbone=dict(type='YOLOv8CSPDarknet',
arch='P5',
last_stage_out_channels=1024,
deepen_factor=0.33,
widen_factor=0.5,
norm_cfg=norm_cfg,
act_cfg=dict(type='SiLU', inplace=True))

![image](https://user-images.githubusercontent.com/130134976/230837786-2ca218f5-cb00-42cf-a6cf-93e70a33f82b.png)
![image](https://user-images.githubusercontent.com/130134976/230838256-9a91e6e2-387c-4c3a-986b-836e08608be8.png)

Reply:
> Can we find a way to make it more user-friendly and automated?

Sorry, I have found that using @AvoidCUDAOOM.retry_if_cuda_oom in mmdet can also solve this issue.